roles/pulp_user/tasks/main.yml

```diff
@@ -0,0 +1,107 @@
+---
+- name: Get user list
+  uri:
+    url: "{{ pulp_user_url }}"
+    user: "{{ pulp_admin_username }}"
+    password: "{{ pulp_admin_password }}"
+    method: GET
+    status_code: 200
+    force_basic_auth: true
+  no_log: true
+  register: users_list_result
+
+- name: Set fact usernames
+  set_fact:
+    usernames: "{{ (usernames | default([])) + [item.username] }}"
+  with_items: "{{ users_list_result.json.results }}"
+
+- name: Create users
+  vars:
+    pulp_users_present_nn: "{{ pulp_users_present if pulp_users_present is iterable else [] }}"
+  uri:
+    url: "{{ pulp_user_url }}"
+    user: "{{ pulp_admin_username }}"
+    password: "{{ pulp_admin_password }}"
+    force_basic_auth: true
+    method: POST
+    status_code: 201
+    body:
+      username: "{{ item.username }}"
+      password: "{{ item.password | default(omit) }}"
+      first_name: "{{ item.first_name | default(omit) }}"
+      last_name: "{{ item.last_name | default(omit) }}"
+      email: "{{ item.email | default(omit) }}"
+      is_staff: "{{ item.is_staff | default(omit) }}"
+      is_active: "{{ item.is_active | default(omit) }}"
```

Comment:
I might be wrong, but I'm not sure if the omits will work here. It evaluates to a placeholder string that Ansible looks for in module arguments.

Reply:
My understanding was that it would just ignore the argument if it hasn't been defined, so nothing is sent. As they're optional in the API, they would just be set to the default value

Reply:
Yes, but omit doesn't leave it undefined, it sets it to a placeholder string like `omit_gjrgsifnglkdnsf\alksnglkgnflk`. Please test & confirm.

Reply:
Mark's correct, omit only works for *module args*, not for arbitrary datastructures.

Reply:
I've been able to run some tests and I'm not seeing the effects of it. With:
```
pulp_users_present:
  - username: alex-test-user-1
    password: correct horse battery staple
    first_name: alex
    last_name:
    groups:
      - ansiblegroup
```
I get:
```
"pulp_href": "/pulp/api/v3/users/68/",
    "id": 68,
    "username": "alex-test-user-1",
    "first_name": "alex",
    "last_name": "",
    "email": "",
    "is_staff": false,
    "is_active": true,
    "date_joined": "2022-11-03T13:27:08.031239Z",
    "groups": [
      {
        "name": "ansiblegroup",
        "pulp_href": "/pulp/api/v3/groups/2/"
      }
    ]
```

But I see what you mean from other posts online. I'm not sure why it isn't setting any of the attributes to the random string in this case

Reply:
It's possible Ansible behaviour has changed here.

Reply:
I think there's some logic to hide it in output:
```yaml
- hosts: localhost
  gather_facts: no
  vars:
    myvar:
      a: aardvark
      b: "{{ omit }}"
      c: camel
  tasks:
    - debug:
        msg: "{{ myvar }}"
    - debug:
        msg: "{{ myvar }}: {{ myvar | length }}"
```

```
PLAY [localhost] ********************************************************************************************************************

TASK [debug] ************************************************************************************************************************
ok: [localhost] => {
    "msg": {
        "a": "aardvark",
        "c": "camel"
    }
}

TASK [debug] ************************************************************************************************************************
ok: [localhost] => {
    "msg": "{'a': 'aardvark', 'b': '__omit_place_holder__26ffe1849923dcd2c0ccd13b8a0ea62a195e18aa', 'c': 'camel'}: 3"
}
```

```
$ ansible --version
ansible [core 2.13.5]
  config file = /home/rocky/caas-slurm-update/ansible.cfg
  configured module search path = ['/home/rocky/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /home/rocky/caas-slurm-update/venv/lib64/python3.8/site-packages/ansible
  ansible collection location = /home/rocky/.ansible/collections:/usr/share/ansible/collections
  executable location = /home/rocky/caas-slurm-update/venv/bin/ansible
  python version = 3.8.12 (default, May 10 2022, 23:46:40) [GCC 8.5.0 20210514 (Red Hat 8.5.0-10)]
  jinja version = 3.1
```

Reply:
Now that is weird. The output I posted was straight from the pulp cli

Reply:
I've done some testing and switched from `omit` to `None`. I don't know if it would ever cause an issue but `None` has all the correct effects without the risk of weirdness.